src/platform/packages/shared/kbn-test/src/jest/run.ts

```diff
@@ -137,31 +148,79 @@ export function runJest(configName = 'jest.config.js') {
     }
 
     log.verbose(`no config provided, found ${configPath}`);
+    resolvedConfigPath = configPath;
     process.argv.push('--config', configPath);
 
     if (!testFilesProvided) {
       log.verbose(`no test files provided, setting to current directory`);
       process.argv.push(relative(wd, cwd));
     }
-
-    log.info('yarn jest', process.argv.slice(2).join(' '));
-
-    if (SCOUT_REPORTER_ENABLED) {
-      // Expose Jest config file path via environment variables
-      process.env.JEST_CONFIG_PATH = configPath;
+  }
+  let initialOptions: Config.InitialOptions | undefined;
+
+  // If a config path was provided via argv, use that; otherwise we already set resolvedConfigPath above
+  if (argv.config) {
+    try {
+      initialOptions = JSON.parse(argv.config);
+    } catch (err) {
+      resolvedConfigPath = argv.config;
     }
   }
 
-  if (process.env.NODE_ENV == null) {
-    process.env.NODE_ENV = 'test';
+  if (!initialOptions && !resolvedConfigPath) {
```

Comment:
I'm not sure how this condition should pan out - maybe you meant `||` instead of `&&`.

Or maybe there's a missed arg reparse before using `argv` in line :162 again after :152 adds a resolved config path to the process.argv list?

Reply:
Addressed this with a more structural refactoring.